nodeJS/express/controllers.md

```diff
@@ -108,75 +108,99 @@ function myMiddleware(req, res, next) {
+
+Then for the controller, create the following file:
 
 ```javascript
-// user controller file - controllers/userController.js
+// controllers/authorController.js
 
-const getUserById = async (req, res) => {
-  const userId = req.params.id;
+const { getAuthorById } = require("../db");
 
-  const user = await someDBQueryToGetUser(userId);
+const getAuthorById = async (req, res) => {
+  const { authorId } = req.params;
 
-  if (!user) {
-    res.status(404).send("User not found");
+  const author = await getAuthorById(authorId);
+
+  if (!author) {
+    res.status(404).send("Author not found");
     return;
   }
 
-  res.send(`User found: ${user.name}`);
+  res.send(author);
 };
+
+module.exports = { getAuthorById };
 ```
 
-In this example, the `getUserById` function is a controller that handles a specific action related to retrieving a user by their ID. You'll use this controller by importing it into file where routes are defined, and using it like so `router.get("/user/:id", getUserById)`. Let's break down what's happening in this controller:
+In this example, the `getAuthorById` function is a controller that handles a specific action related to retrieving an author by their ID. We'll use this controller by importing it into the file where we define our routes, and using it like this: `router.get("/:authorId", getAuthorById)`. Let's break down what's happening in this controller:
 
-1. The controller extracts the userId from the request parameters (`req.params.id`). This assumes that the parameter is defined with a route, such as `/users/:id`.
-1. It then invokes a database query function `someDBQueryToGetUser` to retrieve the user data based on the userId.
-1. If the user is not found (`!user`), the controller sends a response with a 404 status code and the message `User not found`, using `res.status(404).send(...)`. It then returns from the controller function in order to not invoke any other logic in the controller, because sending a response does not stop the function execution itself.
-1. If the user is found, the controller sends a response with a 200 status code and the message `User found: ${user.name}`, using `res.send(...)`. This assumes that the user object has a name property. We don't need to use the `status` method here since the response sends a 200 by default.
+1. The controller extracts the `authorId` from the request parameters (`req.params.authorId`). This assumes we've defined the parameter with a route, such as `/authors/:authorId`.
+1. It then invokes a database query function `getAuthorById` to retrieve the author data based on the `authorId`.
+1. If the controller doesn't find the author, it sends a response with a 404 status code and the message Author not found, using `res.status(404).send(...)`. It then returns from the controller function to avoid invoking any other logic in the controller, as sending a response doesn't automatically stop the function execution.
```

Comment:
```suggestion
1. If the controller doesn't find the author, it sends a response with a 404 status code and the message `"Author not found"`, using `res.status(404).send(...)`. It then returns from the controller function to avoid invoking any other logic in the controller, as sending a response doesn't automatically stop the function execution.
```